services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/S3AccessPointEndpointResolver.java

```diff
@@ -134,13 +134,15 @@ private String validateConfiguration(S3EndpointResolverContext context, S3Resour
                                                + "addressing enabled.");
         }
 
-        if (!isArnRegionEnabled(serviceConfiguration) && clientRegionDiffersFromArnRegion(region, arnRegion)) {
-            throw new IllegalArgumentException(
-                String.format("The region field of the ARN being passed as a bucket parameter to an S3 operation "
-                              + "does not match the region the client was configured with. To enable this "
-                              + "behavior and prevent this exception set 'useArnRegionEnabled' to true in the "
-                              + "configuration when building the S3 client. Provided region: '%s'; client region:"
-                              + " '%s'.", arnRegion, region));
+        if (!isArnRegionEnabled(serviceConfiguration) || isFipsRegion(region.toString())) {
+            if (clientRegionDiffersFromArnRegion(region, arnRegion)) {
```

Comment:
clientRegionDiffersFromArnRegion currently strips "fips" from arn regions as well, but arn regions should never contain fips, correct?

We probably don't want people setting the client region to non-fips, then modifying the ARN region to fips and having it successfully go to a non-fips endpoint.

Reply:
Agree.. Using similar check as [isFipsRegionProvided](https://github.com/aws/aws-sdk-java-v2/blob/584ccb59e770177aeaa4c3b6bda4e24015b8ece9/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/S3EndpointUtils.java#L56-L61) , but I cannot use this method directlty since this method relies on useArnRegion to check if the arn is fips.
And it will fail for the below case
arn.--> fips region client --> non fips  and useArnRegion enabled.